### PR TITLE
bump to kube 1.17

### PR DESCRIPTION
**Pull request by vareti**
This PR aims to rebase this repo with Kubernetes 1.17

### CODE REVIEW

**Comment by vareti** [comment]
Closing this pull request  as it is not needed anymore by [library-go](https://github.com/openshift/library-go/pull/631/files#diff-395f41b2a27b70ce44399c91c82158c2L20)

File: Gopkg.toml [hunk 1]
```diff
@@ -41,31 +41,31 @@
 
 [[constraint]]
   name = "k8s.io/api"
-  version = "kubernetes-1.16.2"
+  version = "kubernetes-1.17.0-rc.2"
```
Comment (caesarxuchao) [inline Gopkg.toml]: We shouldn't use the rc version.

Comment (vareti) [reply Gopkg.toml]: I am in the process of pushing a new commit with `v0.17.0` tag. Will push the changes once I done.

Comment (vareti) [reply Gopkg.toml]: Thanks for reviewing

Comment (vareti) [reply Gopkg.toml]: Thanks for reviewing

**Comment by caesarxuchao** [comment]
The disruptive test is flaky. Created https://github.com/kubernetes-sigs/kube-storage-version-migrator/issues/51.

**Comment by deads2k** [comment]
> The disruptive test is flaky. Created #51.

flaky or broken?

@vareti please open a zero diff change and get some test history on it to compare.

**Comment by vareti** [comment]
> 
> flaky or broken?
> 
> @vareti please open a zero diff change and get some test history on it to compare.

@deads2k  created #52 and will try to get test history

**Comment by sttts** [comment]
Is the last commit intentional or just for debugging CI?

**Comment by sttts** [comment]
Otherwise lgtm.

File: test/e2e/tests/chaos.go [hunk 2]
```diff
@@ -214,6 +215,19 @@ func (t *StorageMigratorChaosTest) Test(done <-chan struct{}) {
 			if len(ss.Status.PersistedStorageVersionHashes) == 1 && ss.Status.PersistedStorageVersionHashes[0] == ss.Status.CurrentStorageVersionHash {
 				continue
 			}
+			listOfMigrations, err := t.migrationClient.MigrationV1alpha1().StorageVersionMigrations().List(metav1.ListOptions{})
+			if err != nil {
+				util.Logf("unable to get list of storage version migrations: %s", err)
+			} else {
+				for _, migration := range listOfMigrations.Items {
```
Comment (deads2k) [inline test/e2e/tests/chaos.go]: nesting this an extra level deep looks questionable to me.  If you want to have this information, why don't you dump it somewhere where you aren't already in a loop of other resources?

**Comment by deads2k** [comment]
fix or remove the last commit.  

/lgtm
/approve
/hold

holding for the debugging commit to be resolved.

**Comment by vareti** [comment]
@sttts @deads2k removed the last commit. it was added to debug disruptive test failure.

**Comment by sanchezl** [comment]
/test pull-kube-storage-version-migrator-disruptive
tests running slow again :-(

**Pull request merged**